File: fhg-api/src/main/java/de/jbamberger/api/FhgApi.java
package de.jbamberger.api;

import android.arch.lifecycle.LiveData;

import de.jbamberger.api.data.VPlan;

/**
 * @author Jannik Bamberger (dev.jbamberger@gmail.com)
 */

public interface FhgApi {

    LiveData<ApiResponse<VPlan>> getVPlan();
}

File: fhg-api/src/main/java/de/jbamberger/api/FhgApi.kt
package de.jbamberger.api

import android.app.Application
import android.arch.lifecycle.LiveData
import de.jbamberger.api.data.FeedItem
import de.jbamberger.api.data.VPlan
import javax.inject.Inject

/**
 * @author Jannik Bamberger (dev.jbamberger@gmail.com)
 */

interface FhgApi {

    class Provider(app: Application) {

        @Inject
        lateinit var api: FhgApi

        init {
            DaggerFhgApiComponent.builder().application(app).build().inject(this)
        }

    }

    val vPlan: LiveData<ApiResponse<VPlan>>

    val feed: LiveData<ApiResponse<List<FeedItem>>>
}
